scipy/special/_hyp2f1.pxd

```diff
@@ -128,10 +132,9 @@ cdef inline double complex hyp2f1_complex(
             max_degree = fabs(a) - 1
         else:
             max_degree = fabs(b) - 1
-        # If number of terms is excessively large, we return nan without
-        # attempting to compute.
-        if max_degree < 15000:
-            # This cast is OK because we've ensured max_degree isn't too large.
+        if max_degree <= INT_MAX:
```

Comment:
This could lead to platform-dependent behavior; in a follow up it might be nice to explicitly use a (u)int64 say (which is ok since we can use C99 now).

Reply:
Sure. Will do